static/src/javascripts/projects/common/modules/onward/history.js

```diff
@@ -0,0 +1,477 @@
+
+const editionalised = [
+    'business',
+    'commentisfree',
+    'culture',
+    'environment',
+    'media',
+    'money',
+    'sport',
+    'technology',
+];
+
+const pageMeta = [
+    {
+        tid: 'section',
+        tname: 'sectionName',
+    },
+    {
+        tid: 'keywordIds',
+        tname: 'keywords',
+    },
+    {
+        tid: 'seriesId',
+        tname: 'series',
+    },
+    {
+        tid: 'authorIds',
+        tname: 'author',
+    },
+];
+
+const buckets = [
+    {
+        type: 'content',
+        indexInRecord: 1,
+    },
+    {
+        type: 'front',
+        indexInRecord: 2,
+    },
+];
+
+const summaryPeriodDays = 90;
+const forgetUniquesAfter = 10;
+const historySize = 50;
+const storageKeyHistory = 'gu.history';
+const storageKeySummary = 'gu.history.summary';
+
+const today = Math.floor(Date.now() / 86400000); // 1 day in ms
+
+let historyCache;
+let summaryCache;
+let popularFilteredCache;
+let topNavItemsCache;
+let inMegaNav = false;
+
+const saveHistory = (history: Array<Array<any>>): void => {
+    historyCache = history;
+    local.set(storageKeyHistory, history);
+};
+
+const saveSummary = (summary: Object): void => {
+    summaryCache = summary;
+    local.set(storageKeySummary, summary);
+};
+
+const getHistory = (): Array<Array<any>> => {
+    historyCache = historyCache || local.get(storageKeyHistory) || [];
+    return historyCache;
+};
+
+const getSummary = (): Object => {
+    if (!summaryCache) {
+        summaryCache = local.get(storageKeySummary);
+
+        if (
+            !isObject(summaryCache) ||
+            !isObject(summaryCache.tags) ||
+            typeof summaryCache.periodEnd !== 'number'
+        ) {
+            summaryCache = {
+                periodEnd: today,
+                tags: {},
+                showInMegaNav: true,
+            };
+        }
+    }
+    return summaryCache;
+};
+
+const seriesSummary = (): Object => {
+    const views = item => item.reduce((acc, record) => acc + record[1], 0);
+
+    const seriesTags = pick(getSummary().tags, (v, k) => k.includes('series'));
+
+    const seriesTagsSummary = mapValues(
+        seriesTags,
+        tag => views(tag[1]) + views(tag[2])
+    );
+
+    return seriesTagsSummary;
+};
+
+const mostViewedSeries = (): string => {
+    const summary = seriesSummary();
+    let series = '';
+
+    Object.keys(summary).forEach(key => {
+        const views = summary[key];
+
+        if (!series || views > summary[series]) {
+            series = key;
+        }
+    });
+
+    return series;
+};
+
+const deleteFromSummary = (tag: string): void => {
+    const summary = getSummary();
+
+    delete summary.tags[tag];
+    saveSummary(summary);
+};
+
+const isRevisit = (pageId: string): boolean => {
+    const visited = getHistory().find(page => page[0] === pageId);
+
+    return !!(visited && visited[1] > 1);
+};
+
+const pruneSummary = (summary: Object, mockToday?: ?number) => {
+    const newToday = mockToday || today;
```

Comment:
sure, if I do this I'll change the argument name from `mockToday` to `newToday` and assign `today` as the default value.